clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -258,7 +258,7 @@ public static void writeVarint(int value, ByteBuffer buffer) {
     public static void writeVarlong(long value, DataOutput out) throws IOException {
         long v = (value << 1) ^ (value >> 63);
         while ((v & 0xffffffffffffff80L) != 0L) {
-            out.writeByte(((int) v & 0x7f) | 0x80);
+            out.writeByte((byte) (v & 0x7f) | 0x80);
```

Comment:
Since the type expected by `writeByte` is `int`, it seems like the previous code was OK? Otherwise we do two conversions.